### DISCUSSION TITLE
Get current logged user in custom endpoint

### DISCUSSION THREAD

**Discussion by Antoine-lb** [strapi/strapi#7453] [Legacy Topics]
Hello,
I created a simple custom controller just like in [the docs for controllers](https://strapi.io/documentation/v3.x/concepts/controllers.html#custom-controllers).

The problem is that I don't hace access to `ctx.state.user` even when I send the `JWT`. Tested with endpoints build in the interface and I do have access to the user, so is just for the custom endpoint.

I most be missing a middleware or something.

This brings me the question of: should I even build an endpoint without the interfeace? I just wanted a simple enpoint that just returns the current user to check if the user is still valid for the auto-loggin feature.

Thanks for the help!

My route:
```
{
  "routes": [
    {
      "method": "POST",
      "path": "/current-user",
      "handler": "current-user.index"
    }
  ]
}
```

My controller:
```
module.exports = {
  // POST /current-user
  index: async (ctx) => {
    const user = ctx.state.user;
    console.log("user", user);
    ctx.send("Hello World!");
  },
};

```

**Comment by MattieBelt** [comment]
@Antoine-lb not sure why the `ctx.state.user` is not working, it should work.

I personally use `GET /users/me `, to check auth and retrieve updated user info.

**Reply by derrickmehaffy** [reply]
MattieBelt is correct here, the better endpoint to use would be the built in `/users/me`

**Reply by Antoine-lb** [reply]
Thanks for the answer, it does exactly what I was searching for. Just to know, where is that in the documentation (a.k.a. how do you know that)?
